Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -117,13 +157,53 @@ class Navigator {
     
     var electronicHorizonOptions: ElectronicHorizonOptions? {
         didSet {
-            let nativeOptions: MapboxNavigationNative.ElectronicHorizonOptions?
-            if let electronicHorizonOptions = electronicHorizonOptions {
-                nativeOptions = MapboxNavigationNative.ElectronicHorizonOptions(electronicHorizonOptions)
-            } else {
-                nativeOptions = nil
-            }
-            navigator.setElectronicHorizonOptionsFor(nativeOptions)
+            setupElectronicHorizonOptions()
+        }
+    }
+}
+
+extension Navigator: FallbackVersionsObserver {
+    
+    enum TileVersionState {
+        /// No tiles version switch is required. Navigator has enough tiles for map matching.
+        case nominal
+        /// Navigator does not have tiles on current version for map matching, but TileStore contains regions with required tiles of a different version
+        case shouldFallback([String])
+        /// Navigator is in a fallback mode but newer tiles version were successefully downloaded and ready to use.
+        case shouldReturnToLatest
+    }
+    
+    func onFallbackVersionsFound(forVersions versions: [String]) {
```

Comment:
What thread this callbacks are called on? Any guarantees from NavNative about the thread?

Reply:
Currently it is called on a worker thread. But I think it's better to do the navigator reset logic on main thread